src/main/java/org/folio/circulation/resources/RequestFromRepresentationService.java

```diff
@@ -150,51 +162,63 @@
   }
 
   private CompletableFuture<Result<Request>> fetchItemAndLoanForPageTlrRequest(Request request) {
-    if (request.getOperation() == Request.Operation.CREATE) {
-      return fromFutureResult(fetchItemForPageTlr(request)
-        .thenApply(r -> r.mapFailure(err -> errorHandler.handleValidationError(err,
-          NO_AVAILABLE_ITEMS_FOR_TLR, r))))
-        .flatMapFuture(this::fetchFirstLoanForUserWithTheSameInstanceId)
-        .flatMapFuture(this::fetchUserForLoan)
-        .toCompletableFuture();
-    } else {
-      return fromFutureResult(fetchItemForRequest(request))
-        .flatMapFuture(req -> succeeded(req).after(loanRepository::findOpenLoanForRequest)
-          .thenApply(r -> r.map(req::withLoan)))
-        .flatMapFuture(this::fetchUserForLoan)
-        .toCompletableFuture();
-    }
+    return request.getOperation() == Request.Operation.CREATE
+      ? fetchDataForPageTlr(request)
+      : fetchDataForNonPageTlr(request);
+  }
+
+  private CompletableFuture<Result<Request>> fetchDataForPageTlr(Request request) {
+    return fromFutureResult(completedFuture(fetchItemForPageTlr(request))
+      .thenApply(r -> r.mapFailure(err -> errorHandler.handleValidationError(err,
+        NO_AVAILABLE_ITEMS_FOR_TLR, r))))
+      .flatMapFuture(this::fetchFirstLoanForUserWithTheSameInstanceId)
+      .flatMapFuture(this::fetchUserForLoan)
+      .toCompletableFuture();
+  }
+
+  private CompletableFuture<Result<Request>> fetchDataForNonPageTlr(Request request) {
+    return fromFutureResult(fetchItemForRequest(request))
+      .flatMapFuture(req -> succeeded(req).after(loanRepository::findOpenLoanForRequest)
+        .thenApply(r -> r.map(req::withLoan)))
+      .flatMapFuture(this::fetchUserForLoan)
+      .toCompletableFuture();
+  }
+
+  private Result<Request> fetchItemForPageTlr(Request request) {
```

Comment:
We usually use "fetch" for cases when we fetch something from another module or database. Here, it looks like, we're just getting it from the list.

Reply:
fixed

Reply:
Hm.. I don't see any changes here